src/cpp/src/visual_language/llava_next/classes.cpp

```diff
@@ -331,29 +332,33 @@ ov::Tensor pack_image_features_llava_next(
 
 } // namespace
 
-ov::Tensor InputsEmbedderLLaVANext::get_inputs_embeds(const std::string& prompt, const std::vector<ov::Tensor>& images, ov::genai::VLMPerfMetrics& metrics) {
-    std::string image_token = m_vlm_config.im_start;
-
+std::vector<ov::genai::EncodedImage> InputsEmbedderLLaVANext::encode_images(const std::vector<ov::Tensor>& images) {
+    std::vector<EncodedImage> embeds;
+    ov::AnyMap vision_config = {{"patch_size", m_vlm_config.vision_config_patch_size}};
     std::vector<ov::Tensor> single_images = to_single_image_tensors(images);
+    for (const ov::Tensor& image : single_images) {
+        embeds.emplace_back(m_vision_encoder->encode(image, vision_config));
+    }
+    return embeds;
+}
+
+ov::Tensor InputsEmbedderLLaVANext::get_inputs_embeds(const std::string& prompt, const std::vector<ov::genai::EncodedImage>& images, ov::genai::VLMPerfMetrics& metrics) {
+    std::string image_token = m_vlm_config.im_start;
 
     std::string formatted_prompt;
     std::vector<ov::Tensor> image_embeds;
-    image_embeds.reserve(single_images.size());
-    
+    image_embeds.reserve(images.size());
     ov::Tensor image_newline;
 
-    for (const auto& image : single_images) {
-        ov::AnyMap vision_config = {{"patch_size", m_vlm_config.vision_config_patch_size}};
-        EncodedImage encoded_image = m_vision_encoder->encode(image, vision_config);
-
+    for (const auto& encoded_image : images) {
         if (!image_newline) {
             size_t embed_dim = encoded_image.resized_source.get_shape().at(2);
             image_newline = ov::Tensor(encoded_image.resized_source.get_element_type(), {embed_dim});
             float* image_newline_data = image_newline.data<float>();
             std::copy(m_vlm_config.image_newline.begin(), m_vlm_config.image_newline.end(), image_newline_data);
         }
 
-        ImageSize original_image_size{image.get_shape().at(1), image.get_shape().at(2)}; // [height, width]
+        ImageSize original_image_size{encoded_image.original_image_size.height, encoded_image.original_image_size.width}; // [height, width]
 
         ov::Tensor packed_features = pack_image_features_llava_next(encoded_image, original_image_size, image_newline);
```

Comment:
`pack_image_features_llava_next()` already takes `encoded_image` as an arg. Don't duplicate the info. Teach `pack_image_features_llava_next()` to extract that values instead.

Reply:
Done.